pkg/controllers/resources/pods/conditions.go

```diff
@@ -0,0 +1,81 @@
+package pods
+
+import (
+	synccontext "github.com/loft-sh/vcluster/pkg/controllers/syncer/context"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// UpdateConditions adds/updates new/old conditions in the physical Pod
+func UpdateConditions(ctx *synccontext.SyncContext, physicalClusterClient kubernetes.Interface, physicalPod *corev1.Pod, virtualPod *corev1.Pod) error {
+	// check if the readinessGates are added to vPod
+	if len(virtualPod.Spec.ReadinessGates) > 0 {
+		newCustomConditions, existingCustomConditions := getCustomConditions(physicalPod, virtualPod)
+		// check if newConditions need to be added.
+		if len(newCustomConditions) > 0 {
+			physicalPod.Status.Conditions = append(physicalPod.Status.Conditions, newCustomConditions...)
+			_, err := physicalClusterClient.CoreV1().Pods(physicalPod.Namespace).UpdateStatus(ctx.Context, physicalPod, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+		}
+
+		// check if existingConditions need to be updated.
+		if len(existingCustomConditions) > 0 {
+			for index, condition := range existingCustomConditions {
+				physicalPod.Status.Conditions[index] = condition
+			}
+			_, err := physicalClusterClient.CoreV1().Pods(physicalPod.Namespace).UpdateStatus(ctx.Context, physicalPod, metav1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// getCustomConditions returns new and existing custom conditions in the given virtual pod.
+func getCustomConditions(physicalPod *corev1.Pod, virtualPod *corev1.Pod) ([]corev1.PodCondition, map[int]corev1.PodCondition) {
+	// newConditions don't require index so slice will suffice.
+	newCustomConditions := []corev1.PodCondition{}
+	// existingConditions will need index so, need a map here.
+	existingCustomConditions := map[int]corev1.PodCondition{}
+	var foundCondition *corev1.PodCondition
+	var index int
+	for _, vCondition := range virtualPod.Status.Conditions {
+		if isCustomCondition(vCondition) {
+			for i, pCondition := range physicalPod.Status.Conditions {
+				// found condition in pPod with same type, updating foundCondition and index
+				if isCustomCondition(pCondition) && vCondition.Type == pCondition.Type {
+					foundCondition = &pCondition
+					index = i
+					break
+				}
+			}
+			if foundCondition != nil {
+				// To avoid same status updates, checking status again.
+				if foundCondition.Status != vCondition.Status {
+					existingCustomConditions[index] = vCondition
+				}
+			} else {
+				// foundCondition is nil means that its a new condition in vPod, needs to be updated to pPod
+				newCustomConditions = append(newCustomConditions, vCondition)
+			}
+			// resetting the helper variables
+			index = -1
+			foundCondition = nil
+		}
+	}
+
+	return newCustomConditions, existingCustomConditions
+}
+
+// Check for customCondition
+func isCustomCondition(condition corev1.PodCondition) bool {
```

Comment:
Check `spec.readinessGates.conditionTypes` (see https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate) and only allow those as custom conditions